### PR TITLE
Continuous deployment instrumentation

### PR DESCRIPTION
**Pull request by kislyuk**
- Use [Chalice](https://github.com/awslabs/chalice) to enable deployment of the app to AWS Lambda/API Gateway
- Add a Chalice-Connexion adapter to preserve the ability to use Connexion validation
- Add a `deploy` config to Travis CI to perform automatic deployment any time we push to master

In our AWS account, I have separately set up the DNS/SSL/CloudFront to serve the app on https://hca-dss.czi.technology/v1/.

### CODE REVIEW

File: dss/__init__.py [hunk 1]
```diff
@@ -17,10 +17,14 @@
 
 logging.basicConfig(level=logging.DEBUG)
 
+app = None
```
Comment (ttung) [inline dss/__init__.py]: why do these need to be globals?

can the request methods just get the current app object via `flask.current_app`?

Comment (kislyuk) [inline dss/__init__.py]: Good point. I didn't know about `flask.current_app`. That works better as a global accessor. I'll delete this one.

Comment (kislyuk) [inline dss/__init__.py]: Deleted.

**Comment by ttung** [comment]
What credentials are required to deploy this?

**Comment by ttung** [comment]
The deploy stage failed. Are you deploying using the credentials embedded in `.travis.yml`?  If so, were the AWS role extended with the proper permissions?

**Comment by kislyuk** [comment]
@ttung - Sorry about the breakage. Yes, I need to audit the AWS role and add permissions according to the needs of the deploy stage. I'll do that later today.

**Comment by kislyuk** [comment]
Should be all fixed now. I committed a few Travis fixes directly on master. I'll add some docs or a setup script for getting the IAM policy set up correctly for the CI IAM user.

**Pull request merged**